modules/sara_r4/cellular_r4.c

```diff
@@ -490,11 +492,58 @@ uint32_t _Cellular_GetSocketId( CellularContext_t * pContext,
         cellularStatus = CELLULAR_BAD_PARAMETER;
     }
 
-    if( ( cellularStatus == CELLULAR_SUCCESS ) &&
-        ( sessionId >= MIN_TCP_SESSION_ID ) && ( sessionId <= MAX_TCP_SESSION_ID ) )
+    if( ( cellularStatus == CELLULAR_SUCCESS ) && ( sessionId <= MAX_TCP_SESSION_ID ) )
```

Comment:
The condition ( sessionId >= MIN_TCP_SESSION_ID ) can't be ignored.

Reply:
```
#define MIN_TCP_SESSION_ID          ( 0U )
uint32_t _Cellular_GetSocketId( CellularContext_t * pContext,
                                uint8_t sessionId )
```
uint8_t is always >= 0

Reply:
But now I see #define MIN_TCP_SESSION_ID          ( 0 ) instead of #define MIN_TCP_SESSION_ID          ( 0U )
One of them need to be corrected.

Reply:
Sorry for typo it is 
```
#define MIN_TCP_SESSION_ID          ( 0 )
```
uint8_t is always  >= 0. What is the point to compare uint8_t >= 0?

```
./cellular_r4_api.c:451:                if( ( tempValue >= MIN_TCP_SESSION_ID ) && ( tempValue <= MAX_TCP_SESSION_ID ) )
./cellular_r4_urc_handler.c:395:                if( ( tempValue >= MIN_TCP_SESSION_ID ) && ( tempValue <= MAX_TCP_SESSION_ID ) )
./cellular_r4_urc_handler.c:490:                if( ( tempValue >= MIN_TCP_SESSION_ID ) && ( tempValue <= MAX_TCP_SESSION_ID ) )
./cellular_r4_urc_handler.c:554:                if( ( tempValue >= MIN_TCP_SESSION_ID ) && ( tempValue <= MAX_TCP_SESSION_ID ) )
```
The usage here is to compare int32_t ( result of Cellular_ATStrtoi ).

Reply:
Original warning message.
```
if( ( cellularStatus == CELLULAR_SUCCESS ) &&
        ( sessionId >= MIN_TCP_SESSION_ID ) && ( sessionId <= MAX_TCP_SESSION_ID ) )
```

Reply:
Update the type comparison MAX_TCP_SESSION_ID in next commit.